mirgecom/gas_model.py

```diff
@@ -309,47 +355,97 @@ def make_fluid_state(cv, gas_model, temperature_seed=None,
-        smoothness_beta=smoothness_beta
-    )
-
-    from mirgecom.eos import MixtureEOS, MixtureDependentVars
-    if isinstance(gas_model.eos, MixtureEOS):
-        dv = MixtureDependentVars(
-            temperature=dv.temperature,
-            pressure=dv.pressure,
-            speed_of_sound=dv.speed_of_sound,
-            smoothness_mu=dv.smoothness_mu,
-            smoothness_kappa=dv.smoothness_kappa,
-            smoothness_beta=dv.smoothness_beta,
-            species_enthalpies=gas_model.eos.species_enthalpies(cv, temperature))
+    if wall_vars is None:
+        temperature = gas_model.eos.temperature(cv=cv,
+                                                temperature_seed=temperature_seed)
+        pressure = gas_model.eos.pressure(cv=cv, temperature=temperature)
+
+        if limiter_func:
+            cv = limiter_func(cv=cv, pressure=pressure, temperature=temperature,
+                              dd=limiter_dd)
+
+        dv = GasDependentVars(
+            temperature=temperature,
+            pressure=pressure,
+            speed_of_sound=gas_model.eos.sound_speed(cv, temperature),
+            smoothness_mu=smoothness_mu,
+            smoothness_kappa=smoothness_kappa,
+            smoothness_beta=smoothness_beta
+        )
+
+        from mirgecom.eos import MixtureEOS, MixtureDependentVars
+        if isinstance(gas_model.eos, MixtureEOS):
+            dv = MixtureDependentVars(
+                temperature=dv.temperature,
+                pressure=dv.pressure,
+                speed_of_sound=dv.speed_of_sound,
+                smoothness_mu=dv.smoothness_mu,
+                smoothness_kappa=dv.smoothness_kappa,
+                smoothness_beta=dv.smoothness_beta,
+                species_enthalpies=gas_model.eos.species_enthalpies(cv, temperature)
+            )
+
+        if gas_model.transport is not None:
+            tv = gas_model.transport.transport_vars(cv=cv, dv=dv, eos=gas_model.eos)
+            return ViscousFluidState(cv=cv, dv=dv, tv=tv)
+
+        return FluidState(cv=cv, dv=dv)
+
+    else:
 
-    if gas_model.transport is not None:
-        tv = gas_model.transport.transport_vars(cv=cv, dv=dv, eos=gas_model.eos)
-        return ViscousFluidState(cv=cv, dv=dv, tv=tv)
+        tau = gas_model.wall.eval_tau(wall_vars)
+        epsilon = gas_model.wall._sample_model.void_fraction(tau)
+        temperature = \
+            gas_model.wall.get_temperature(cv=cv, wv=wall_vars,
+                                           tseed=temperature_seed, tau=tau,
+                                           eos=gas_model.eos)
 
-    return FluidState(cv=cv, dv=dv)
+        pressure = (
+            1.0/epsilon*gas_model.eos.pressure(cv=cv, temperature=temperature))
+
+        if limiter_func:
+            # FIXME this is something that may give trouble in the future. Lets see.
+            cv = limiter_func(cv=cv, pressure=pressure, temperature=temperature,
+                              epsilon=epsilon, dd=limiter_dd)
```

Comment:
Hrm.  Would there be any trouble with just calling this first thing upon entry to this function regardless of whether `wall_vars` is present? Then we'd be using limited CV throughout.

Reply:
The thing is that we need the "epsilon" (the void fraction/volume filled by the gas inside the wall). That's why I added the "FIXME" so I dont forget about it when actually solving the "NS eqs" inside the wall.

Anyway, epsilon=1.0 for non-porous flow so the answer to your question is "Maybe".